runtime/src/main/java/org/corfudb/runtime/view/SnowflakeGuidFactory.java

```diff
@@ -0,0 +1,49 @@
+package org.corfudb.runtime.view;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Fast Globally Unique Identity (GUID) generator that returns compact ids
+ * having a notion of a universal comparable ordering.
+ *
+ *    ______________________________________________
+ *   | timestamp lower 42 bits | Node Id | Sequence |
+ *    ------------------------- --------- ----------
+ *    <------ 42 bits ---------><-10bits-><-12bits-->
+ *
+ * Created by Sundar Sridharan on 5/22/19.
+ */
+public class SnowflakeGuidFactory implements AbstractOrderedGUIDFactory {
+    public static final short NODE_MAX_VAL = 1023;
+    public static final int NODE_BIT_POS = 10;
+    public static final short SEQ_MAX_VAL = 4095;
+    public static final int SEQ_MSB = 12;
+
+    private short node;
+    private AtomicInteger sequence;
+
+    /**
+     * Use this node id to scope the generator to a namespace.
+     * For example:
+     *      UUID nodeId = UUID.fromString("mac_address_of_machine");
+     *      SnowflakeGuidFactory guidFactory = new SnowflakeGuidFactory(nodeId.node());
+     * @param node
+     */
+    public SnowflakeGuidFactory(long node) {
```

Comment:
Should this class be a singleton class and `nexGUID()` be a static method? Also if the node() is a machine specific mac address, why not let this class generate it instead of pass it here?

Reply:
I did not know of an easy way to generate it. But I think you are right, let me attempt your suggestion instead.

Reply:
I don't agree, never use static, always try to avoid using it

Reply:
Or if you really want to use static method in this case then rename SnowflakeGuidUtil and don't se interface

Reply:
Actually I cannot make it a singleton because it is meant to be scoped to a corfu runtime. If we ever support multiple runtime in the same process, then they should have their own namespaces. Otherwise it is already singleton.

Reply:
> Actually I cannot make it a singleton because it is meant to be scoped to a corfu runtime. If we ever support multiple runtime in the same process, then they should have their own namespaces. Otherwise it is already singleton.

Then I think you should instantiate this  factory class when the runtime is created, instead of instantiating it every time you need to generate a GUID, right? Or you can make `nexGUID()` a static method, to make this class sth. like util class.

Reply:
Done. I have included the instantiation into the runtime and modified the tests accordingly.